src/hooks/useJobList.ts

```diff
@@ -0,0 +1,109 @@
+import { useMemo, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useInfiniteQuery } from 'react-query';
+import { useApi } from '../hooks/useApi';
+
+export interface Job {
+    id: string;
+    title: string;
+    company: string;
+    city: string;
+    modality: string;
+    typeContract: string;
+    type: string;
+    contractText?: string;
+    description: string;
+    indefinideContract: boolean;
+    salaryMin?: number;
+    salaryMax?: number;
+    benefits?: string;
+    prerequisites: string;
+    federalUnit: string;
+    createdAt: string;
+    company_id: string;
+}
+
+export const useJobList = () => {
+    const [selectedJob, setSelectedJob] = useState<string | null>('');
+    const [clickedJob, setClickedJob] = useState<Job[] | Job>([]);
```

Comment:
Nesse caso, poderíamos sempre tratar como um array de Jobs e fazemos as tratativas necessárias na posição 0 quando for um elemento só, o que acha?